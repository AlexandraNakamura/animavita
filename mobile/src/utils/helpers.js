import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = array => array instanceof Array && array.length === 0;

export const getToken = async () => JSON.parse(await AsyncStorage.getItem('@animavita:authToken')) || {};
