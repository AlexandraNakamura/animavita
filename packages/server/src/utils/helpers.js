import AWS from 'aws-sdk';

require('dotenv').config();

const {
  S3_KEY, S3_SECRET, S3_BUCKET, S3_REGION
} = process.env;

export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      translate: 'Macho'
    },
    {
      key: 'female',
      translate: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).translate;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      translate: 'Pequeno'
    },
    {
      key: 'medium',
      translate: 'Médio'
    },
    {
      key: 'bigger',
      translate: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).translate;
};

export const uploadFile = async (upload) => {
  let url = '';
  // eslint-disable-next-line new-cap
  const base64Data = new Buffer.from(
    upload.preview.uri.replace(/^data:image\/\w+;base64,/, ''),
    'base64'
  );

  const params = {
    Bucket: S3_BUCKET,
    Key: upload.image.name,
    Body: base64Data,
    ACL: 'public-read',
    ContentEncoding: 'base64',
    ContentType: upload.image.type
  };

  const s3 = new AWS.S3({
    accessKeyId: S3_KEY,
    secretAccessKey: S3_SECRET,
    region: S3_REGION
  });

  try {
    const { Location } = await s3.upload(params).promise();
    url = Location;
  } catch (err) {
    throw err;
  }

  return url;
};
