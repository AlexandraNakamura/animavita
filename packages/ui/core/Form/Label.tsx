import React from 'react';
import styled from 'styled-components/native';
import {useTheme} from '@animavita/theme/index';

import Typography, {TypographyProps} from '../Typography';

// styles
export const Container = styled.View``;

const Form: React.FC<TypographyProps> = ({...props}) => {
  const theme = useTheme();
  return <Typography variant="body" color={theme.styledTheme.grey} {...props} />;
};

export default Form;
