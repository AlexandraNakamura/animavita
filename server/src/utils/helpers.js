export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      trans: 'Macho'
    },
    {
      key: 'female',
      trans: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).trans;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      trans: 'Pequeno'
    },
    {
      key: 'medium',
      trans: 'Médio'
    },
    {
      key: 'bigger',
      trans: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).trans;
};
