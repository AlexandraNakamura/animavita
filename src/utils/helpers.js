import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = (array) => {
  if (array instanceof Array && array.length === 0) {
    return true;
  }

  return false;
};

export const getUser = async () => JSON.parse(await AsyncStorage.getItem('@animativa:user'));
