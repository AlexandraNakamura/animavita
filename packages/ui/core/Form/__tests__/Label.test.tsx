import React from 'react';
import renderer from 'react-test-renderer';

import {ThemeContext} from '../../../tests/helpers';
import Label from '../Label';

describe('Label', () => {
  it('renders correctly', () => {
    const tree = renderer.create(ThemeContext(<Label>Test</Label>)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
