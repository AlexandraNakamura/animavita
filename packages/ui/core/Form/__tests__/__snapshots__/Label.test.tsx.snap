// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label renders correctly 1`] = `
<Text
  color="#BABABA"
  style={
    Array [
      Object {
        "color": "#353434",
      },
      Object {
        "color": "#BABABA",
        "fontSize": 26.5,
      },
    ]
  }
  themeName="light"
>
  Test
</Text>
`;
